### ISSUE TITLE
Reliable way to lock up vscode

### ISSUE THREAD

**Issue by NinoFloris** [OmniSharp/omnisharp-vscode#1088]
## Environment data
`dotnet --info` output:
.NET Command Line Tools (1.0.0-preview2-1-003177)

Product Information:
 Version:            1.0.0-preview2-1-003177
 Commit SHA-1 hash:  a2df9c2576

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.11
 OS Platform: Darwin
 RID:         osx.10.11-x64
VS Code version:
1.8.1
C# Extension version:
1.6.1

## Steps to reproduce

Set file to C# and paste this in, next try to add any generic type to the int on _totalLockup.

```cs
using System;
using System.Collections.Generic;

public class Bizarre 
{
    // Typing X< (say List<) before x (int here) works fine when TKey is not RuntimeTypeHandle
    private Dictionary<Type, int> _worksFine;

    // With RuntimeTypeHandle this fully locks up VSCode until it shows the 'window unresponsive' dialog
    private Dictionary<RuntimeTypeHandle, int> _totalLockup;
}
```

See tweet for repro https://twitter.com/NinoFloris/status/812838424545783808

**Comment by DustinCampbell** [comment]
This looks like a duplicate of https://github.com/OmniSharp/omnisharp-vscode/issues/1086.

**Comment by DustinCampbell** [comment]
OK. All of my analysis is in #1086. I'm closing this as a dupe of this issue. However, thanks *very* much for your excellent repro. I've pushed a new release ([1.6.2](https://github.com/OmniSharp/omnisharp-vscode/releases/tag/v1.6.2)) that should address this problem. It's not a perfect solution, but it won't hang any longer. If you're interested in seeing what happened, check out issue #1086 and the PR I merged to workaround it: #1089.
